datastore.py

```diff
@@ -36,16 +33,9 @@ def data(self):
                 self.__data = {}
         return self.__data
      
-    @data.setter
-    def data(self, value):
-        """Saves the data to S3 on assignment, for example the following code
-        will automatically serialize the python dict to json and save to S3.
-        
-        ds = Datastore()
-        ds.data = {"a": "b"}
-
-        Args:
-            value (object): arbitrary data to be serialized and stored to S3

```

Comment:
Just out of curiosity, why is there a change from a setter method to a save method?  Does it have to do with making the switch from Heroku to S3?  Is it a best practice issue to use a save with S3 as opposed to a setter?

Reply:
like i mentioned on fb, you’re commenting on an outdated piece of code. check the “Files changed” tab for the latest changes.

As for why i flip-flopped from setter property to a save method, it has nothing to do with heroku or S3. It’s more because I felt that the setter syntax obfuscates the fact that it’s making a network call. a network call that takes time a non-trivial amount of time, and can fail based on network connection.

E.g. `ds.data = {“a”: “b”}` looks innocuous, but it’s really trying to save a file in s3. i figure an explicit `save()` method is more obvious.